crates/xilem_web/src/context.rs

```diff
@@ -41,12 +41,52 @@ fn remove_attribute(element: &web_sys::Element, name: &str) {
     }
 }
 
+fn set_class(element: &web_sys::Element, class_name: &str) {
+    #[cfg(debug_assertions)]
+    if class_name.is_empty() {
+        panic!("class names cannot be the empty string");
+    }
+    #[cfg(debug_assertions)]
+    if class_name.contains(' ') {
+        panic!("class names cannot contain the ascii space character");
+    }
+    element.class_list().add_1(class_name).unwrap_throw()
```

Comment:
[Something, I have observed](https://www.measurethat.net/Benchmarks/Show/30272/0/classname-vs-classlist-with-existing-classes), is that `className` seems to be (significantly) faster than `classList`. Since we have full control over which classes that element uses, it maybe makes sense to use `className` instead and concatenate all the classes when a change occurs there?

Reply:
I propose that se use `classList` to start with, then do an experiment where we compare the two approaches (and then use the fastest).